pkg/registry/core/pod/strategy.go

```diff
@@ -60,7 +61,8 @@ func (podStrategy) NamespaceScoped() bool {
 func (podStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
 	pod := obj.(*api.Pod)
 	pod.Status = api.PodStatus{
-		Phase: api.PodPending,
+		Phase:    api.PodPending,
+		QOSClass: qos.InternalGetPodQOS(pod),
```

Comment:
I'm pretty sure the unit test would basically be the same as `TestGetPodQOS()` in `qos_test.go`.  I can work on an e2e test.